presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -133,38 +136,18 @@ private AggregationAnalyzer(List<Expression> groupByExpressions, Scope sourceSco
         this.expressions = groupByExpressions.stream()
                 .map(e -> ExpressionTreeRewriter.rewriteWith(new ParameterRewriter(analysis.getParameters()), e))
                 .collect(toImmutableList());
-        ImmutableList.Builder<Integer> fieldIndexes = ImmutableList.builder();
-
-        fieldIndexes.addAll(groupByExpressions.stream()
-                .filter(FieldReference.class::isInstance)
-                .map(FieldReference.class::cast)
-                .map(FieldReference::getFieldIndex)
-                .iterator());
-
-        // For a query like "SELECT * FROM T GROUP BY a", groupByExpressions will contain "a",
-        // and the '*' will be expanded to Field references. Therefore we translate all simple name expressions
-        // in the group by clause to fields they reference so that the expansion from '*' can be matched against them
-        List<Expression> nonFieldReferenceGroupingExpressions = expressions.stream()
-                .filter(expression -> analysis.getColumnReferences().contains(expression) && !(expression instanceof FieldReference))
-                .collect(toImmutableList());
-        for (Expression expression : nonFieldReferenceGroupingExpressions) {
-            QualifiedName name;
-            if (expression instanceof Identifier) {
-                name = QualifiedName.of(((Identifier) expression).getName());
-            }
-            else {
-                name = DereferenceExpression.getQualifiedName((DereferenceExpression) expression);
-            }
 
-            List<Field> fields = sourceScope.getRelationType().resolveFields(name);
-            checkState(fields.size() <= 1, "Found more than one field for name '%s': %s", name, fields);
+        this.columnReferences = analysis.getColumnReferenceFields();
 
-            if (fields.size() == 1) {
-                Field field = Iterables.getOnlyElement(fields);
-                fieldIndexes.add(sourceScope.getRelationType().indexOf(field));
-            }
-        }
-        this.fieldIndexes = fieldIndexes.build();
+        this.groupingFields = groupByExpressions.stream()
+                .map(columnReferences::get)
```

Comment:
not a problem at all. I don't like the idea of depending on nulls (especially in maps).